### DISCUSSION TITLE
New HTML Tag Processor - An in-depth tutorial

### DISCUSSION THREAD

**Discussion by bph** [WordPress/developer-blog-content#75] [Topic Ideas]
WordPress 6.2 introduces the first component in a [new HTML processing API](https://github.com/WordPress/gutenberg/issues/44410).  Before WordPress 6.2, the block markup was typically updated using regular expressions or the DOMDocument class. Both have downsides. The former is tedious and prone to security issues, while the latter uses libxml2 which [does not support HTML5](https://gitlab.gnome.org/GNOME/libxml2/-/issues/211). ([Adam Zielinski](https://github.com/WordPress/gutenberg/issues/44410#issuecomment-1420787816)) 

[Beyond the Dev Note](https://github.com/WordPress/gutenberg/issues/44410#issuecomment-1420787816), it would be beneficial to PHP developers if there is a longer in-depth post on the Developer Blog. 

cc: @adamziel , @dmsnell, @justintadlock

**Comment by justintadlock** [comment]
Just noting that I'm interested in taking this on if no one else has the bandwidth for it. I know there's been some previous work done, so I don't want to take over if there are already plans for it.

**Reply by dmsnell** [reply]
@adamziel has been building an interactive post and he and I have been preparing for an interview-format discussion about it.

Glad to have you join if you want @justintadlock - we haven't been posting here the updates on what we're doing.

I'll let @adamziel drop a link if he wants to/can if you want to review the material.

**Reply by adamziel** [reply]
Thanks for the ping @dmsnell !

@justintadlock I published an interactive tutorial on my blog yesterday:

https://adamadam.blog/2023/02/16/how-to-modify-html-in-a-php-wordpress-plugin-using-the-new-tag-processor-api/

Once the _Interactive PHP_ block becomes reusable I would love to open-source it and bring the same format to learn.wp.org. For now, though, everything is hardcoded to work with this specific topic and on my specific blog. 

@dmsnell and I are also cooking an interview-like discussion as he mentioned.

**Comment by bph** [comment]
This topic has been approved in today's meeting of the editorial group  (March 2nd, 2023) 
I locked this discussion. For further discussion please comment on the issue 

- #83

**Comment by bph** [comment]
Published [The HTML API: process your tags, not your pain](https://developer.wordpress.org/news/2023/09/the-html-api-process-your-tags-not-your-pain/)
